code/_onclick/hud/parallax.dm

```diff
@@ -24,24 +24,32 @@ var/list/parallax_on_clients = list()
 
 /obj/screen/plane_master/parallax_master
 	plane = PLANE_SPACE_PARALLAX
+	blend_mode = BLEND_MULTIPLY
 	color = list(
 	1,0,0,0,
 	0,1,0,0,
 	0,0,1,0,
 	0,0,0,0,
 	0,0,0,1)
 
+/obj/screen/plane_master/parallax_spacemaster //Turns space white, causing the parallax to only show in areas with opacity. Somehow
+	plane = PLANE_SPACE_BACKGROUND
+	color = list(
+	0,0,0,0,
+	0,0,0,0,
+	0,0,0,0,
+	0,0,0,0,
+	1,1,1,1)
+
 /obj/screen/plane_master/parallax_dustmaster
 	plane = PLANE_SPACE_DUST
 
 /datum/hud/proc/update_parallax()
 	var/client/C = mymob.client
 	if(!parallax_initialized || C.updating_parallax) return
 
-	for(var/turf/T in range(get_turf(C.eye),C.view))
-		if(istype(T,/turf/space))
-			C.updating_parallax = 1
-			break
+	if(locate(/turf/space) in range(get_turf(C.eye),C.view))
```

Comment:
`locate() in trange()` is the fastest of everything mentioned here so far, and by sheer coincidence happens to be about as fast as I said it was before
It's almost exactly 90% faster than `locate() in range()` when there is no space nearby, and about 75% faster when there is only space nearby, and presumably always between those figures
Looping through `trange()` is slower than I said before, though